### PR TITLE
Change ExtractMethod return value to TextDocumentEdits

### PR DESCRIPTION
**Pull request by BladeMF**

### CODE REVIEW

File: lib/Adapter/WorseReflection/Refactor/WorseExtractMethod.php [hunk 1]
```diff
@@ -53,8 +58,71 @@ public function __construct(Reflector $reflector, BuilderFactory $factory, Updat
...
+            count($startNode->statements) > 0
+        ) {
+            $stmt = current($startNode->statements);
+            assert($stmt instanceof Node);
+            while ($stmt && $stmt->getStart() < $offsetStart) {
+                $stmt = next($startNode->statements);
+            }
+            
+            if ($stmt === false) {
+                dump($startNode->getStart(), $startNode->getEndPosition(), $offsetStart);
```
Comment (dantleech) [inline lib/Adapter/WorseReflection/Refactor/WorseExtractMethod.php]: diump

Comment (dantleech) [reply lib/Adapter/WorseReflection/Refactor/WorseExtractMethod.php]: shoudl this return false?

**Review by dantleech — CHANGES_REQUESTED** [review]

File: lib/Adapter/WorseReflection/Refactor/WorseExtractMethod.php (hunk 1 above)
Comment (BladeMF) [reply lib/Adapter/WorseReflection/Refactor/WorseExtractMethod.php]: What do you mean?

Comment (dantleech) [reply lib/Adapter/WorseReflection/Refactor/WorseExtractMethod.php]: what should this do, should it `return false`

Comment (BladeMF) [reply lib/Adapter/WorseReflection/Refactor/WorseExtractMethod.php]: No, I was trying to see when does this case occur. The point of the validation is to ensure that the start and end offsets are in the same code block: 
```php
{
  ...
  <>
  ...
  <>
  ...
}
```
and there are multiple cases related to that depending on where the selection starts and ends. That case concerns the case "empty block", e.g.:
```php
if {
  <>
}
```
The the start statement is the block itself. It's a valid statement to be extracted, so it doesn't need to return `false`.

**Pull request merged**